cmd/orch-tester/orch_tester.go

```diff
@@ -549,3 +599,209 @@ func (s *streamerClient) getOrchestrators() ([]*orch, error) {
+	sourceURI := fmt.Sprintf("%v_0_0/source.m3u8", mid)
+	sourcePl, err := s.downloadMediaPlaylist(sourceURI)
+	if err != nil {
+		return nil, nil, "", err
+	}
+	renditionPl, err := s.downloadMediaPlaylist(mpl.Variants[rand.Int()%len(mpl.Variants)].URI)
+	if err != nil {
+		return nil, nil, "", err
+	}
+
+	if len(renditionPl.Segments) == 0 {
+		return nil, nil, "", fmt.Errorf("no transcoded segments")
+	}
+
+	var rendSegs []*m3u8.MediaSegment
+	var sourceSegs []*m3u8.MediaSegment
+	for _, v := range renditionPl.Segments {
+		if v != nil {
+			rendSegs = append(rendSegs, v)
+		}
+	}
+	for _, v := range sourcePl.Segments {
+		if v != nil {
+			sourceSegs = append(sourceSegs, v)
+		}
+	}
+	if len(rendSegs) == 0 || len(sourceSegs) == 0 {
+		return nil, nil, "", fmt.Errorf("no segments found")
+	}
+
+	seqNo := r % int(math.Min(float64(len(rendSegs)), float64(len(sourceSegs))))
+	rendition, err = s.downloadSegment(fmt.Sprintf("%v/%v", s.broadcaster, rendSegs[seqNo].URI))
+	if err != nil {
+		return nil, nil, "", err
+	}
+	source, err = s.downloadSegment(fmt.Sprintf("%v/%v", s.broadcaster, sourceSegs[seqNo].URI))
+	if err != nil {
+		return nil, nil, "", err
+	}
+	return source, rendition, rendSegs[seqNo].URI, err
+}
+
+func (s *streamerClient) downloadSegment(url string) ([]byte, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	res, err := s.client.Do(req)
+	if err != nil {
+		glog.Error(err)
+		return nil, err
+	}
+	body, err := ioutil.ReadAll(res.Body)
+	defer res.Body.Close()
+	if err != nil {
+		glog.Error(err)
+		return nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		return nil, fmt.Errorf("%v", string(body))
+	}
+	return body, nil
+}
+
+func (s *streamerClient) downloadMasterPlaylist(manifestID string) (*m3u8.MasterPlaylist, error) {
+	url := fmt.Sprintf("%v/stream/%v_0_0.m3u8", s.broadcaster, manifestID)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := s.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		return nil, errors.New(string(body))
+	}
+
+	mpl := m3u8.NewMasterPlaylist()
+	if err := mpl.Decode(*bytes.NewBuffer(body), true); err != nil {
+		return nil, err
+	}
+
+	return mpl, nil
+}
+
+func (s *streamerClient) downloadMediaPlaylist(uri string) (*m3u8.MediaPlaylist, error) {
+	url := fmt.Sprintf("%v/stream/%v", s.broadcaster, uri)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := s.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		return nil, errors.New(string(body))
+	}
+
+	pl, err := m3u8.NewMediaPlaylist(1, 1)
```

Comment:
This will always return a windowed playlist with a single segment. Should we use the same [live list length](https://github.com/livepeer/go-livepeer/blob/14b0bd6d92f4b840ffb1e16b3d438640648e36a6/core/playlistmanager.go#L18) as the broadcaster?

Reply:
On the line below we decode the buffer into the playlist instance, wouldn't this expand the window size to what the buffer contains and since this would be the broadcaster's window size of 6 we will use that ?

Reply:
I'm not sure. Could you confirm that this is the case by doing a simple test?

Reply:
Not, it will not. To decode manifest with unknown length this should be used:
https://github.com/livepeer/stream-tester/blob/bfc1bf03f93a003f58913154ba8a945cf2a1177e/internal/testers/m3utester2.go#L563-L574

Reply:
This returns a `Playlist` type instead of a `MediaPlaylist` type though how can we get a `MediaPlaylist` ? 
Decode it line by line ?

Reply:
https://github.com/livepeer/stream-tester/blob/bfc1bf03f93a003f58913154ba8a945cf2a1177e/internal/testers/m3utester2.go#L959-L966